rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/VersionedModelControllerV3.scala

```diff
@@ -148,25 +156,42 @@ abstract class VersionedModelControllerV3[C <: VersionedModel with Product
     } else if (version != item.version) {
       Future.failed(new IllegalArgumentException(s"URL and payload version mismatch: ${version} != ${item.version}"))
     } else {
-      versionedModelService.update(user.getUsername, item).map {
-        case Some((updatedEntity, validation)) =>
-          createdWithNameVersionLocationBuilder(updatedEntity.name, updatedEntity.version, request, stripLastSegments = 2).body(validation)
-        case None => throw notFound()
+      versionedModelService.isDisabled(item.name).flatMap { isDisabled =>
+        if (isDisabled) {
+          throw EntityDisabledException(s"Entity ${item.name} is disabled. Enable it first to create new versions.")
+        } else {
+          versionedModelService.update(user.getUsername, item).map {
+            case Some((updatedEntity, validation)) =>
+              createdWithNameVersionLocationBuilder(updatedEntity.name, updatedEntity.version, request, stripLastSegments = 2).body(validation)
+            case None => throw notFound()
+          }
+        }
       }
     }
   }
 
-  @DeleteMapping(Array("/{name}", "/{name}/{version}"))
+  @PutMapping(Array("/{name}"))
   @ResponseStatus(HttpStatus.OK)
-  def disable(@PathVariable name: String,
-              @PathVariable version: Optional[String]): CompletableFuture[UpdateResult] = {
-    val v = if (version.isPresent) {
-      // For some reason Spring reads the Optional[Int] param as a Optional[String] and then throws ClassCastException
-      Some(version.get.toInt)
-    } else {
-      None
+  def enable(@PathVariable name: String): CompletableFuture[DisabledPayload] = {
+    versionedModelService.enableEntity(name).map { updateResult => // always enabling all version of the entity
+      if (updateResult.getMatchedCount > 0) {
+        DisabledPayload(disabled = false)
+      } else {
+        throw NotFoundException(s"No versions for entity $name found to be enabled.")
+      }
+    }
+  }
+
+  @DeleteMapping(Array("/{name}"))
+  @ResponseStatus(HttpStatus.OK)
+  def disable(@PathVariable name: String): CompletableFuture[DisabledPayload] = {
```

Comment:
I get that we block disabling if there is something using it. And I get that we need to show user what is wrong.
 But getting back a list `used-in` in this call might be weird and unexpected. Can we have something like 

```json
{
  "disabled": "true/false",
  "errors": ["whatever you need (like used-in) in here"]
}
```

Reply:
Well, I think it ok, because you can only expect to receive the expected object/message type on success - the `DisabledPayload` here. 
On error, you may get different response types and based on what is wrong, you get different things - e.g. 404 (no payload or message), 400 - a `UsedIn`, a `Validation` etc.

Reply:
![image](https://user-images.githubusercontent.com/17467768/168077162-5595d40a-92c5-4691-aa84-f62e1f2c3e15.png)

I will be honest. It took me a long time to get what is happening. What am I seeing and why I only understood it from the code. 

Purely logical it might make sense but it is not user friendly and I vote we do it a tad differently. That is add a message and then payload or something.

Reply:
Wrapper with error message added.